### DISCUSSION TITLE
Request to add MQTT SSL certificate

### DISCUSSION THREAD

**Discussion by lumloong** [meatpiHQ/wican-fw#62] [General]
Request to add MQTT SSL certificate
The OBD bus is not secure over HTTP.

**Comment by meatpiHQ** [comment]
@eflydata I don't see the need for secure MQTT, since WiCAN is mostly used on a private network and local broker.

**Reply by lumloong** [reply]
But when my car is parked somewhere, the WiCAN network connects to the car WIFI, and connecting to the home MQTT via `mqtt://` non-SSL exposes an insecure connection.

**Reply by meatpiHQ** [reply]
@eflydata You have a valid point. The issue is that secure MQTT requires certificates, so we'll need to allow upload from the webpage and store the files, it's a bit complicated. I'll add this feature when I have some free time, it's not an easy change.

**Reply by thommy2312** [reply]
Hello, any news on this topic. I use MQTT TLS in Tasmota and it works fine without a certificate on the Tasmota device. 
<img width="397" height="724" alt="image" src="https://github.com/user-attachments/assets/ff3e796c-af33-4525-ae9c-7909a6679e8d" />

**Reply by meatpiHQ** [reply]
It's using a public CA.

I'm actually working on the TLS support right now for the PRO version both https and mqtts.

**Reply by thommy2312** [reply]
Thats great, I have order today a PRO dongle.
